scripts/GWA_TBX/PG04/PG04_WaterQualityParameters_01_IdePix.py

```diff
@@ -0,0 +1,62 @@
+#Definition of inputs and outputs
+#==================================
+##BC=group
+##PG04_WaterQualityParameters_CoastColour_L1P=name
+##ParameterBoolean|Icol|Icol|false
+##ParameterBoolean|Calibration|Calibration|false
+##ParameterBoolean|Smile|Smile|true
+##ParameterBoolean|Equalization|Equalization|true
+##ParameterBoolean|IgnoreSeaIceClim|Ignore sea ice climatology|false
+##ParameterNumber|CloudBufferWidth|lnsert size of cloud buffer in pixel|0|None|2
+##ParameterNumber|CloudScreeningAmbiguous|Cloud screening ambiguous threshold|0|None|1.4
+##ParameterNumber|CloudScreeningSure|Cloud screening sure threshold|0|None|1.8
+##ParameterNumber|GlintCloudThresholdAddition|Glint cloud screening addition|0|None|0.1
+#ParameterBoolean|OutputCloudProbabilityFeatureValue|Write cloud probability feature value|false
+
+import os
+import glob
+import tempfile
+
+tempfolder = 'wq_scripts_'
+OutputCloudProbabilityFeatureValue = 'false'
+
+def folder_create(tempfolder):
+    try:
+        tempdir = glob.glob(os.path.join(tempfile.gettempdir(), tempfolder + '*'))[0]
+        return tempdir
+    except:
+        progress.setConsoleInfo('Temporary folder:' + tempfolder + ' does not exist and will be created.')
+        tempfile.mkdtemp(prefix=tempfolder)
+        tempdir = glob.glob(os.path.join(tempfile.gettempdir(), tempfolder + '*'))[0]
+        return tempdir
+
+def folder_check(tempfolder):
+    try:
+        tempdir = glob.glob(os.path.join(tempfile.gettempdir(), tempfolder + '*'))[0]
+        return False
+    except IndexError:
+        progress.setConsoleInfo('ERROR: Temporary folder:' + tempfolder + ' cloud not be created. Check for administration rights to create folder.')
+        return True
+
+def create_parameterfile(tempdir, Icol, Calibration, Smile, Equalization, IgnoreSeaIceClim, CloudBufferWidth, CloudScreeningAmbiguous, CloudScreeningSure, GlintCloudThresholdAddition, OutputCloudProbabilityFeatureValue):
+    with open(tempdir + "WaterQualityParameters01.txt", "w") as text_file:
+        text_file.write('icol='+ str(Icol).lower() + '\n')
```

Comment:
It would be much easier for you and cleaner for the code if you automated the part of writing all these lines:

```python
def create_parameterfile(tempdir, **kwargs):
    paramfile = os.path.join(tempdir, 'WaterQualityParameters01.txt')
    with open(paramfile, 'w') as f:
        for key, value in kwargs.items():
            keylower = key[0].lower() + key[1:]
            f.write('{}={}\n'.format(keylower, str(value).lower()))
```

Reply:
Could you explain a little what you mean with automate? The current modules translate GUI parameters set by the user, into a parameter file read by gpt.

Reply:
> Could you explain a little what you mean with automate?

I meant to reduce the repetition of lines with exactly the same statements but just different words. Loops and dictionaries make this much more compact.

Reply:
@j08lue if you want to do that, please go ahead. ;) I am not a software developer. Therefore it was a lot learning by doing. In this current version of the code it was easy to keep track of all the parameter inputs even though there are some repetitions. But I totally get your point. With more time on my hands I would have done some cleaning and improvements.